tests/drizzle/test_kernel.py

```diff
@@ -0,0 +1,138 @@
+import pytest
+import numpy as np
+from astropy import wcs
+import cdriz_setup
+
+
+@pytest.fixture
+def kernel_pars():
+    _params = cdriz_setup.Get_Grid(inx=50, iny=60, outx=51, outy=66)
+    _params.zero_background()
+    return _params
+
+
+@pytest.mark.parametrize("kernel", ["square", "point", "turbo", "gaussian", "lanczos3"])
+def test_point_kernel(kernel, kernel_pars, return_png=False):
+    """Function tests different c code point kernels (inputs already created on instantiation).
+
+    Parameters
+    ----------
+    kernel : str
+        String associated with one of the c code point kernel options.
+    kernel_pars : Class
+        The Class inintialized in Get_Grid which includes all of the inputs need to run cdriz.tdriz.
+    return_png: bool, optional
+        Flag, whether to create an output image map (png).
+    """
+    # truth filename
+    truth_filename = f"./tests/drizzle/truth_files/{kernel}_truth"
```

Comment:
Same comment here (and all the truth_filename construction code) with regard to still needing this much of a path name to the file.

Reply:
I guess I'm not sure how to make this shorter using the _file_ variable without moving the truth files. There may be a way, but I might need help getting there.

Reply:
For the failing tests, you need to do the following with the the truth_filename:

# truth filename
    truth_filename = f"truth_files/{kernel}_truth".     === No slash at the beginning and minimum path hardcoded.

    # Check to see if the specified file exists in the current working directory 
    print(f"path: {os.path.dirname(__file__)}")   === You must execute pytest as “pytest -s drizzle” to see this message
    if not os.path.exists(truth_filename):  === This will be false
        # Find where this module has been installed and determine the absolute path
        install_dir = os.path.dirname(__file__) === This is /Users/mdelapena/repos/drizzlepac/tests/drizzle/
        default_file = os.path.join(install_dir, truth_filename) === Constructs the full name (without the csv)
        truth_filename = default_file
        print(f"truth: {truth_filename}")

This is just so you can get the nearly full truth filename (no csv suffix) and have only the minimum filename hardcoded.  In particular, I sometimes want to run a subset of tests in the subdirectory.  I would copy, for example, the desired tests from the drizzle subdirectory to “mydrizzle” and then “pytest -s mydrizzle”.  The tests would fail because “drizzle” has been hardcoded to the truth filename.

Also, is there a reason not to have the “.csv” included in the original truth_filename?

Reply:
Ahh okay, understood. I left the ".csv" off to pass the filename to both the save_png (.png) and save_file (.csv) functions.

Reply:
I'm pretty sure that I understand what you were suggesting I do. 

I've created a new function in cdriz_setup that I use for the rest of the tests. The function takes in a relative path and a name, joins them both to each other and the absolute path, and then prints and returns the final path.